run_matrix.py

```diff
@@ -0,0 +1,168 @@
+    """Clean the directory used to store repos for the tests."""
+    if root.is_dir():
+        shutil.rmtree(root)
+
+
+def _generate_test(interface: str, test_path: Path, fixture_id: str) -> Path:
+    """Generate a pytest file for a given charm and interface."""
+    test_content = textwrap.dedent(f"""from interface_tester import InterfaceTester
+    def test_{interface}_interface({fixture_id}: InterfaceTester):
+        {fixture_id}.configure(
+        interface_name="{interface}",
+    )
+    {fixture_id}.run()
+    """)
+    test_filename = f"interface-test-{interface}.py"
+    with open(test_path / test_filename, "w") as file:
+        file.write(test_content)
+    return test_path / test_filename
+
+
+def _get_fixture(charm_config: _CharmTestConfig, charm_path: Path) -> FixtureSpec:
+    """Get the tester fixture from a charm."""
+    fixture_path = charm_path / FIXTURE_PATH
+    fixture_id = FIXTURE_IDENTIFIER
+    if charm_config.test_setup:
+        if charm_config.test_setup["location"]:
+            fixture_path = charm_path / Path(charm_config.test_setup["location"])
+        if charm_config.test_setup["identifier"]:
+            fixture_id = charm_config.test_setup["identifier"]
+    return FixtureSpec(fixture_path, fixture_id)
+
+
+def _setup_venv(charm_path: Path) -> Path:
+    """Create the venv for a charm and return the path to its python."""
+    logging.info(f"Installing dependencies in venv for {charm_path}")
+    original_wd = os.getcwd()
+    os.chdir(charm_path)
+    # Create the venv and install the requirements
+    try:
+        subprocess.check_call("python -m venv ./.interface-venv", shell=True, stdout=subprocess.DEVNULL)
+        subprocess.check_call(
+            ".interface-venv/bin/python -m pip install setuptools pytest git+https://github.com/canonical/interface-tester-pytest@main",
+            shell=True,
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL
+        )
+        subprocess.check_call(
+            ".interface-venv/bin/python -m pip install -r requirements.txt",
+            shell=True,
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL
+        )
+    except subprocess.CalledProcessError as e:
+        raise SetupError("venv setup failed") from e
+    os.chdir(original_wd)
+    return charm_path / ".interface-venv/bin/python"
+
+
+def test_charm(charm_path: Path, test_path: Path):
+    """Run the interface test for a given charm and interface."""
+    logging.info(f"Running tests for {charm_path}")
+    original_wd = os.getcwd()
+    os.chdir(charm_path)
+    try:
+        subprocess.check_call(
+            f"PYTHONPATH=src:lib .interface-venv/bin/python -m pytest {test_path}",
+            shell=True
+        )
+    except subprocess.CalledProcessError as e:
+        raise InterfaceTestError from e
+    os.chdir(original_wd)
+
+
+def run_interface_tests(path: Path, include: str = "*", clean=True):
```

Comment:
Why the `clean` and `include` parameters? `run_interface_tests` is only called once without those parameters

Reply:
removed as unused for now, we will add them back later